src/hooks/usePathRewrite.ts

```diff
@@ -17,6 +22,15 @@ const usePathRewrite = () => {
     if (Array.isArray(safe)) safe = safe[0]
     if (!safe) return
 
+    const { prefix } = parsePrefixedAddress(safe)
+
+    const chainId = Object.entries(chains).find(([key]) => key === prefix)?.[1]
```

Comment:
Chain id can be also passed as its own query param.

Reply:
Should we check for that case here? i.e. we try to open `/bogus:0x00...?chain=gor` imo it should still navigate to the welcome page

Reply:
I mean, it can be _just_ the chain param. E.g. `/welcome?chain=bogus`.
`useChainId` handles both.

Reply:
Correct it will now stay on the last opened chain / default chain but still display the `?chain=bogus` query param where previously it would crash the app. Should we remove the chain query param if its not a valid chain? Afaik we are not preserving it when navigating through the app.